lib/trino-geospatial-toolkit/src/test/java/io/trino/geospatial/rtree/BenchmarkJtsStrTreeQuery.java

```diff
@@ -0,0 +1,115 @@
+import org.openjdk.jmh.annotations.OperationsPerInvocation;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import org.openjdk.jmh.runner.options.VerboseMode;
+
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static io.trino.geospatial.rtree.RtreeTestUtils.makeRectangles;
+
+@State(Scope.Thread)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(2)
+@Warmup(iterations = 3, time = 2, timeUnit = TimeUnit.SECONDS)
+@Measurement(iterations = 10, time = 1, timeUnit = TimeUnit.SECONDS)
+@BenchmarkMode(Mode.AverageTime)
+public class BenchmarkJtsStrTreeQuery
+{
+    private static final int SEED = 613;
+    private static final int NUM_PROBE_RECTANGLES = 1000;
+
+    @Benchmark
+    @OperationsPerInvocation(NUM_PROBE_RECTANGLES)
+    public void rtreeQuery(BenchmarkData data, Blackhole blackhole)
+    {
+        for (Envelope query : data.getProbeEnvs()) {
+            data.getRtree().query(query, blackhole::consume);
+        }
+    }
+
+    @State(Scope.Thread)
+    public static class BenchmarkData
+    {
+        @Param({"1000", "3000", "10000", "30000", "100000", "300000", "1000000"})
+        private int numBuildRectangles;
+
+        private List<Envelope> probeEnvs;
+        private STRtree rtree;
+
+        @Setup
+        public void setup()
+        {
+            Random random = new Random(SEED);
+            probeEnvs = makeRectangles(random, NUM_PROBE_RECTANGLES).stream()
+                    .map(rect -> new Envelope(rect.getXMin(), rect.getXMax(), rect.getYMin(), rect.getYMax()))
+                    .collect(Collectors.toList());
+            List<Envelope> buildEnvs = makeRectangles(random, numBuildRectangles).stream()
+                    .map(rect -> new Envelope(rect.getXMin(), rect.getXMax(), rect.getYMin(), rect.getYMax()))
+                    .collect(Collectors.toList());
+            rtree = buildRtree(buildEnvs);
+        }
+
+        public List<Envelope> getProbeEnvs()
+        {
+            return probeEnvs;
+        }
+
+        public STRtree getRtree()
+        {
+            return rtree;
+        }
+
+        private STRtree buildRtree(List<Envelope> envs)
+        {
+            STRtree rtree = new STRtree();
+            for (Envelope env : envs) {
+                rtree.insert(env, env);
+            }
+            rtree.build();
+            return rtree;
+        }
+    }
+
+    public static void main(String[] args)
+            throws Throwable
+    {
+        Options options = new OptionsBuilder()
```

Comment:
For all of the benchmark `main(String[] args)` methods, Trino has switched to using `BenchmarkBuilder` instance instead of this style of runner. You'll want to check out some other benchmark classes in Trino to see how to adapt those.

Reply:
Done